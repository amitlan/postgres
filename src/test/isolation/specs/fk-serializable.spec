

setup
{
 CREATE TABLE pk (a int PRIMARY KEY);
 CREATE TABLE fk (a int REFERENCES pk ON DELETE CASCADE ON UPDATE CASCADE);
 INSERT INTO pk VALUES (1);
}

teardown
{
 DROP TABLE fk, pk;
}

session "s1"

setup	{ BEGIN ISOLATION LEVEL SERIALIZABLE; }
step "s1spk" { TABLE fk; }
step "s1dpk" { DELETE FROM pk; }
step "s1upk" { UPDATE pk SET a = a + 1; }
step "s1c"	{ COMMIT; }


session "s2"

step "s2ifk" { INSERT INTO fk VALUES (1); }

permutation "s1spk" "s2ifk" "s1dpk" "s1c"
permutation "s1spk" "s2ifk" "s1upk" "s1c"
