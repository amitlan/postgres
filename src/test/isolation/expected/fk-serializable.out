Parsed test spec with 2 sessions

starting permutation: s1spk s2ifk s1dpk s1c
step s1spk: TABLE fk;
a              

step s2ifk: INSERT INTO fk VALUES (1);
step s1dpk: DELETE FROM pk;
ERROR:  could not serialize access due to concurrent update
step s1c: COMMIT;

starting permutation: s1spk s2ifk s1upk s1c
step s1spk: TABLE fk;
a              

step s2ifk: INSERT INTO fk VALUES (1);
step s1upk: UPDATE pk SET a = a + 1;
ERROR:  could not serialize access due to concurrent update
step s1c: COMMIT;
